docs/user/ppl/admin/datasources.rst

```diff
@@ -25,44 +25,91 @@ Definitions of datasource and connector
 
 Example Prometheus Datasource Definition ::
 
-    [{
+    {
         "name" : "my_prometheus",
         "connector": "prometheus",
         "properties" : {
             "prometheus.uri" : "http://localhost:8080",
             "prometheus.auth.type" : "basicauth",
             "prometheus.auth.username" : "admin",
             "prometheus.auth.password" : "admin"
-        }
-    }]
+        },
+        "allowedRoles" : ["prometheus_access"]
+    }
 Datasource configuration Restrictions.
 
 * ``name``, ``connector``, ``properties`` are required fields in the datasource configuration.
+* In case of secure domains, ``allowedRoles`` can be used to specify the opensearch roles allowed to access the datasource via PPL/SQL.
+* In case of secure domains, the plugin would only provide access to admin by default for PPL/SQL queries on datasources.
 * All the datasource names should be unique and match the following regex[``[@*A-Za-z]+?[*a-zA-Z_\-0-9]*``].
 * Allowed Connectors.
     * ``prometheus`` [More details: `Prometheus Connector <prometheus_connector.rst>`_]
 * All the allowed config parameters in ``properties`` are defined in individual connector pages mentioned above.
 
-Configuring a datasource in OpenSearch
+Datasource configuration Management.
 ======================================
+Datasource configuration can be managed using below REST APIs.
 
-* Datasources are configured in opensearch keystore as secure settings under ``plugins.query.federation.datasources.config`` key as they contain credential info.
-* A json file containing array of datasource configurations should be injected into keystore with the above mentioned key. sample json file can be seen in the above section.
+* Datasource Creation POST API ("_plugins/_query/_datasources") ::
 
+    POST https://localhost:9200/_plugins/_query/_datasources
+    content-type: application/json
+    Authorization: Basic {{username}} {{password}}
```

Comment:
Why not use http and no auth to explain this behavior? Do we want to emphasize something?

Reply:
is it not preferred? I am talking about authorizations down below, so I thought using secure domain endpoints would make things clear.